pkg/lib/synthesize/non_inferrable_csv.go

```diff
@@ -0,0 +1,72 @@
+package synthesize
+
+import (
+	"fmt"
+	"strings"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	apiVer "github.com/operator-framework/api/pkg/lib/version"
+	"github.com/operator-framework/api/pkg/operators/v1alpha1"
+)
+
+type NonInferableCSV struct {
+	Name           string                 `json:"name"`
+	Version        apiVer.OperatorVersion `json:"version"`
+	MinKubeVersion string                 `json:"minKubeVersion"`
+	InstallModes   []v1alpha1.InstallMode `json:"installModes"`
+	Description    string                 `json:"description,omitempty"`
+	DisplayName    string                 `json:"displayName,omitempty"`
+	Keywords       []string               `json:"keywords,omitempty"`
+	Labels         map[string]string      `json:"labels,omitempty" `
+	Links          []v1alpha1.AppLink     `json:"links,omitempty"`
+	Maintainers    []v1alpha1.Maintainer  `json:"maintainers,omitempty"`
+	Maturity       string                 `json:"maturity,omitempty"`
+	Provider       v1alpha1.AppLink       `json:"provider,omitempty"`
+	Selector       *metav1.LabelSelector  `json:"selector,omitempty"`
+	Icon           []v1alpha1.Icon        `json:"icon,omitempty"`
+	Replaces       string                 `json:"replaces,omitempty"`
+	Annotations    map[string]string      `json:"annotations,omitempty"`
+	Descriptions   []Description          `json:"descriptions,omitempty"`
+}
+
+type Description struct {
+	Name              string                          `json:"name"`
+	Version           string                          `json:"version"`
+	Kind              string                          `json:"kind"`
+	DisplayName       string                          `json:"displayName,omitempty"`
+	Description       string                          `json:"description,omitempty"`
+	Resources         []v1alpha1.APIResourceReference `json:"resources,omitempty"`
+	StatusDescriptors []v1alpha1.StatusDescriptor     `json:"statusDescriptors,omitempty"`
+	SpecDescriptors   []v1alpha1.SpecDescriptor       `json:"specDescriptors,omitempty"`
+	ActionDescriptor  []v1alpha1.ActionDescriptor     `json:"actionDescriptors,omitempty"`
```

Comment:
These three types are identical. My preference would be to reduce them into a single `Descriptor` type that has a `type` field to discriminate between status, spec, action, etc descriptors. IMO, having a single `descriptors` field in the metadata file for each provided API looks cleaner. Does anyone else have thoughts on this?

(normally, I wouldn't think this is a big deal, but since we're effectively exposing a new public API to end-users, I think we should be extra mindful)

Reply:
Let's do that!